addons/io_scene_gltf/node.py

```diff
@@ -42,26 +42,19 @@ def convert_quaternion(q):
     return Quaternion([q[3], q[0], q[1], q[2]])
 
 
-def get_transform(node):
+def set_transform(node, obj):
     if 'matrix' in node:
-        return convert_matrix(node['matrix'])
+        obj.matrix_local = convert_matrix(node['matrix'])
```

Comment:
This won't affect the user-visible loc/rot/scale so I preferred to decompose the matrix first eg like [this](https://github.com/scurest/gltf-blender-importer/blob/empties/addons/io_scene_gltf/node.py#L4-L25) (but you don't need the coordinate change since you do it with a root object).

Reply:
Agree, had that on my agenda as well.